app/src/androidTest/java/ca/dal/cs/athletemonitor/athletemonitor/AccountManagerTest.java

```diff
@@ -0,0 +1,315 @@
+     * username.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void setUserLoggedInTrueTest() throws Exception {
+       //create a test user and set them as online
+       User testUser = TestingHelper.createTestUser();
+
+       AccountManager.setUserLoginState(testUser.getUsername(), true);
+
+        //attempt to retrieve a reference to the logged in user
+        FirebaseDatabase database = FirebaseDatabase.getInstance();
+        DatabaseReference usersReference = database.getReference("online_users/" + testUser.getUsername());
+
+        //attach a listener for data changes of the users reference.  this will occur when
+        //the reference is populated
+        usersReference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                //if the reference exists, then the user is tagged as logged in
+                if (dataSnapshot.exists()) {
+                    assertTrue(true);
+                } else {
+                    assertTrue(false);
+                }
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        });
+    }
+
+    /**
+     * Tests that a user that has been authenticated is tagged as logged in
+     *
+     * Will pass if there exists a child node in the online_users reference equal to the users
+     * username.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void setUserLoggedInFalseTest() throws Exception {
+        //create a test user and set them as online
+        User testUser = TestingHelper.createTestUser();
+
+        //set the user as online then set them as offline
+        AccountManager.setUserLoginState(testUser.getUsername(), false);
+
+        //attempt to retrieve a reference to the logged in user
+        FirebaseDatabase database = FirebaseDatabase.getInstance();
+        DatabaseReference usersReference = database.getReference("online_users/" + testUser.getUsername());
+
+        //attach a listener for data changes of the users reference.  this will occur when
+        //the reference is populated
+        usersReference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                //if the reference exists, then the user is tagged as logged in and we should fail
+                //the test, otherwise it should succeed
+                if (dataSnapshot.exists()) {
+                    assertFalse(true);
+                } else {
+                    assertFalse(false);
+                }
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        });
+    }
+
+
+
+//    @Test
+//    public void signOutTest() {
```

Comment:
Please uncomment method signOutTest (if it's failing and should pass in the future) or delete (if useless).

Reply:
Removed, not relevant for this user story.